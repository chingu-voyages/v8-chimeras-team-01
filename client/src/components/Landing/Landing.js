import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import BrowseGames from '../Games/BrowseGames.js';
import './index.css';

export default function Landing() {

  /**
   * @method [useState(view)]
   * @description [Creates and sets state for which view is being displayed. This dictates which view is shown and which button is 'chosen']
   */
  const [view, setView] = useState("info");

  /**
   * @function [buttonSwitch]
   * @param {string} selection
   * @description [Changes which view button has sel based on view in state. Used in onClick on icons. VIEW CHANGED ADD/REMOVING CLASSES THROUGH TERNARIES AT CLASSNAMES IN JSX BELOW]
   */
  const buttonSwitch = (selection) => {
    if (selection !== view) {
      setView(selection);
    };
  }

  /**
   * @function [toggleInfo]
   * @description [show/hide info drop down on when screen is larger than 600px]
   */
  const toggleInfo = (boolean) => {
    let geContainer = document.getElementsByClassName("ge-container");
    geContainer[0].classList.toggle("shift-up");
  }

  /**
   * @method [useState(showGames)]
   * @description [Create and change state to show/hide Games component]
   */
  const [showGames, setShowGames] = useState(false);

  /**
   * @function [toggleGames]
   * @param {boolean} boolean
   * @description [Trigger show/hide of Games component]
   */
  const toggleGames = (boolean) => {
    if (boolean !== showGames) {
      setShowGames(boolean);
    }
  }

  return (
    <section id="landing pm0">
      <i className="fas fa-info-circle fa-2x pointy lrg-scrn-icon"
        onClick={toggleInfo}></i>

      <section id="view-button-container" className="pm0">
        <div className={(view === "info") ? "button-box fbc sel" : "button-box fbc"}>
          <i className="fas fa-info-circle fa-2x"
            onClick={() => { buttonSwitch("info") }}></i>
        </div>
        <div className={(view === "game") ? "button-box fbc sel" : "button-box fbc"}>
          <i className="fas fa-gamepad fa-2x"
            onClick={() => { buttonSwitch("game") }}></i>
        </div>
      </section>

      <section className="view-container pm0">

        {/* TERNARY to show/hide in mobile view */}
        <div className={(view === "info") ? "ge-container pm0 shift-up" : "ge-container pm0 shift-left shift-up"}>
          <button className="info-close pm0 pointy"
            onClick={toggleInfo}> X close </button>
          <div className="gameplay-example">
            <div className="fbc">
              <h4 className="ge-header fbc">How do I earn Bragging Rights?</h4>
            </div>
            <div className="ge-box b-s pm0 geb1">
              <div className="ge-description b-s fbc">Setup a game with a few clicks</div>
            </div>
            <div className="ge-box b-s pm0 geb2">
              <div className="ge-description b-s fbc">Answer the questions as they come</div>
            </div>
            <div className="ge-box b-s pm0 geb3">
              <div className="ge-description b-s fbc">See who won Bragging Rights</div>
            </div>
          </div>
        </div>

        {/* TERNARY to show/hide in mobile view */}
        <section className={(view === "game") ? "start-game-container pm0" : "start-game-container pm0 shift-right"}>
          <div className="start-game-section fbc">
            <h4 className="fbc">Quick Launch</h4>
            <Link to="/host/games" className="link fbc">
              <button className="start-game-button fbc pointy">
                <p className="pm0"><strong>Are you a GameMaster??</strong></p>
                <p className="pm0">Host A Game</p>
              </button>
            </Link>
            <Link to="/player/join" className="link fbc">
              <button className="start-game-button fbc pointy">
                <p className="pm0"><strong>Have a Code??</strong></p>
                <p className=" pm0">Join A Game</p>
              </button>
            </Link>
            <button className="browse-button fbc pointy"
              onClick={() => { toggleGames(true) }}>
              <p className="pm0">Browse Games</p>
            </button>
          </div>

          <div className="hth-container pm0">
            <div className="how-to-host">
              <h4 className="hth-header pm0 fbc">Want to Host a Game?</h4>
              <div className="how-to-box htb1 b-s">
                <span className="htb-num">1</span>
                <p>Choose "GameMaster" Option Above</p>
              </div>
              <div className="how-to-box htb2 b-s">
                <span className="htb-num">2</span>
                <p>Select A Game</p>
              </div>
              <div className="how-to-box htb3 b-s">
                <span className="htb-num">3</span>
                <p>Invite Friends!</p>
              </div>
            </div>
          </div>
        </section>

        <BrowseGames showGames={showGames}
          handleToggleGames={toggleGames} />

      </section>

    </section>
  )
}
