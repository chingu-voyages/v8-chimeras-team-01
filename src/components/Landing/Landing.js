import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './index.css';

export default function Landing() {

  /**
   * @method [useState(view)]
   * @description [Creates and sets state for which view is being displayed. This dictates which view is shown and which button is 'chosen']
   */
  const [view, setView] = useState("info");

  /**
   * @function [buttonSwitch]
   * @description [Changes which view button has sel based on view in state. Used in on click]
   */
  const buttonSwitch = (selection) => {
    if (selection !== view) {
      let buttons = document.getElementsByClassName("button-box");
      buttons[0].classList.toggle("sel");
      buttons[1].classList.toggle("sel");
      setView(selection);
      toggleView(selection);
    };
  }

  /**
   * @function [toggleView]
   * @description [Add/Remove shift class to slide correct view into screen]
   */
  const toggleView = (selection) => {
    let gameContainer = document.getElementsByClassName("start-game-container");
    let geContainer = document.getElementsByClassName("ge-container");
    if (selection === "info") {
      gameContainer[0].classList.add("shift-right");
      geContainer[0].classList.remove("shift-left");
    } else if (selection === "game") {
      gameContainer[0].classList.remove("shift-right");
      geContainer[0].classList.add("shift-left");
    }
  }

  return (
    <section id="landing">
      <section id="view-button-container">
        <div className="button-box fbc sel">
          <i className="fas fa-question fa-3x"
             onClick={() => {buttonSwitch("info")}}></i>
        </div>
        <div className="button-box fbc">
          <i className="fas fa-gamepad fa-3x"
             onClick={() => {buttonSwitch("game")}}></i>
        </div>
      </section>

      <section className="view-container">

        <div className="ge-container">
          <div className="gameplay-example">
            <div className="fbc">
              <h4 className="ge-header fbc">How do I earn Bragging Rights?</h4>
            </div>
            <div className="ge-box geb1">
              <div className="ge-description fbc">Setup a game with a few clicks</div>
            </div>
            <div className="ge-box geb2">
              <div className="ge-description fbc">Answer the questions as they come</div>
            </div>
            <div className="ge-box geb3">
              <div className="ge-description fbc">See who won Bragging Rights</div>
            </div>
          </div>
        </div>

        <section className="start-game-container shift-right">
          <div className="start-game-section">
            <h4 className="fbc">Quick Launch</h4>
            <button className="start-game-button fbc">
              <Link to="/games" className="link fbc">
                <p className="button-words"><strong>Are you a GameMaster??</strong></p>
                <p className="button-words">Host A Game</p>
              </Link>
            </button>
            <button className="start-game-button fbc">
              <Link to="/join" className="link fbc">
                <p className="button-words"><strong>Have a Code??</strong></p>
                <p className="button-words">Join A Game</p>
              </Link>
            </button>
          </div>

          <div className="hth-container">
            <div className="how-to-host">
              <h4 className="hth-header fbc">Want to Host a Game?</h4>
              <div className="how-to-box htb1">
                <span className="htb-num">1</span>
                <p>Choose "GameMaster" Option Above</p>
              </div>
              <div className="how-to-box htb2">
                <span className="htb-num">2</span>
                <p>Select A Game</p>
              </div>
              <div className="how-to-box htb3">
                <span className="htb-num">3</span>
                <p>Invite Friends!</p>
              </div>
            </div>
          </div>
        </section>

      </section>

    </section>
  )
}
