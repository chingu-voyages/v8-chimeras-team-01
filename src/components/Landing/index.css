@import '../../App.css';

#landing {
  grid-column: c1 / c2;
  grid-row: r2 / r3;
  width: 100%;
  min-height: calc(100vh - 150px);
}

/*  BUTTONS TO START GAME */

.start-game-section {
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  max-height: 300px;
  padding: 0;
  margin: 1em auto;
}

.start-game-button {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  width: 300px;
  padding: 0;
  margin: 1em auto;
  background-color: var(--color-main);
  background-image:linear-gradient(
    to top right,
    var(--color-gray),
    var(--color-dark) 20%,
    var(--color-main)
    );
  color: #fff;
  border: 2px solid #000;
  border-radius: 1em;
  font-family: var(--font);
  font-size: 1em;
}

.start-game-button:hover {
  background-color: #fff;
  color: #000;
}

.link {
  width: 300px;
  height: 74px;
  padding: 0;
  margin: 0;
}

.button-words {
  padding: 0;
  margin: 0;
}

/* HOW TO HOST GAME */

.hth-container {
  width: 100%;
  background-color: var(--color-main);
}

.how-to-host {
  display: grid;
  grid-template-columns:[htc1] 4fr [htc2] 1fr [htc3] 4fr [htc4];
  grid-template-rows: [htr1] 50px [htr2] 4fr [htr3] 1fr [htr4] 3fr [htr5] 1fr [htr6] 4fr [htr7];
  box-sizing: border-box;
  width: 100%;
  height: 400px;
  padding: 1em;
  margin: 0 auto;
}

.hth-header {
  grid-column: htc1 / htc4;
  grid-row: htr1 / htr2;
  width: 100%;
  height: 30px;
  padding: 0;
  margin: 0;
  font-family: var(--font-header);
  font-size: 1.1em;
  background-color: #fff;
  color: #000;
}

.how-to-box {
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  color: #000;
  box-shadow: 2px 2px 10px #000;
  text-align: center;
}

.how-to-box>p {
  padding-left: 1.5em;
}

.htb-num {
  position: absolute;
  top: 0;
  left: 0px;
  padding: 0.1em;
  margin: 0;
  font-size: 3.5em;
  color: #fff;
  background-color: var(--color-main);
  background-image: url("https://www.transparenttextures.com/patterns/dark-stripes-light.png");
}

.htb1 {
  grid-column: htc1 / htc3;
  grid-row: htr2 / htr4;
  margin-left: 1em;
}

.htb2 {
  grid-column: htc2 / htc4;
  grid-row: htr3 / htr6;
}

.htb3 {
  grid-column: htc1 / htc3;
  grid-row: htr5 / htr7;
}

/* GAMEPLAY EXAMPLE */

.ge-container {
  width: 100%;
  background-color: #fff;
}

.gameplay-example {
  display: grid;
  grid-template-columns:[gec1] 1fr [gec4];
  grid-template-rows: [gerheader] 50px [ger1] 180px [ger2] 20px [ger3] 160px [ger4] 20px [ger5] 180px [ger6];
  box-sizing: border-box;
  width: 100%;
  max-width: 600px;
  height: 600px;
  padding: 0;
  margin: 0 auto;
}

.ge-header {
  grid-column: gec1 / gec4;
  grid-row: gerheader / ger1;
  width: 100%;
  padding: 0;
  margin: 0;
  background-color: #fff;
}

.geb1 {
  grid-column: gec1 / gec2;
  grid-row: ger1 / ger3;
}

.geb2 {
  grid-column: gec1 / gec2;
  grid-row: ger2 / ger5;
}

.geb3 {
  grid-column: gec1 / gec2;
  grid-row: ger4 / ger6;
}

.ge-box {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: flex-start;
  width: 100%;
  padding: 0;
  margin: 0;
  background-color: var(--color-main);
  color: #fff;
  box-shadow: 0 -2px 10px var(--color-gray);
}

.ge-description {
  width: 130px;
  height:130px;
  text-align: center;
  padding: 0;
  margin-left: 1em;
  font-size: 1.1em;
  font-variant: small-caps;
  background-color: #fff;
  color: #000;
  box-shadow: 0 -1px 12px var(--color-gray);
  border-radius: 50%;
}

.geb2>div {
  align-self: flex-end;
  margin: 0 1em 0 0;
}


/*  MEDIA QUERIES  */

@media only screen and (min-width:650px) {

  /*  BUTTONS TO START GAME */

  .start-game-section {
    flex-flow: row nowrap;
    max-width: 650px;
    height: 200px;
  }

  .start-game-button {
    height: 100px;
  }

  .link {
    height: 100px;
  }

  /* HOW TO HOST GAME */

  .how-to-host {
    max-width: 650px;
    height: 600px;
  }

  .hth-header {
    font-size: 1.8em;
  }

  .how-to-box>p {
    font-size: 1.5em;
  }

  .htb-num {
    font-size: 4em;
  }

  /* GAMEPLAY EXAMPLE */

  .gameplay-example {
    display: grid;
    grid-template-columns:[gec1] 1fr [gec4];
    grid-template-rows: [gerheader] 50px [ger1] 220px [ger2] 220px [ger3] 220px [ger4];
    grid-gap: 30px;
    box-sizing: border-box;
    width: 100%;
    height: 850px;
    padding: 0;
    margin: 1em auto;
  }

  .ge-header {
    font-size: 1.8em;
  }

  .geb1 {
    grid-column: gec1 / gec2;
    grid-row: ger1 / ger2;
  }

  .geb2 {
    grid-column: gec1 / gec2;
    grid-row: ger2 / ger3;
  }

  .geb3 {
    grid-column: gec1 / gec2;
    grid-row: ger3 / ger4;
  }

  .ge-box {
    position: relative;
    height: 180px;
  }

  .ge-description {
    position: absolute;
    width: 190px;
    height: 190px;
    padding: 0.5em;
    font-size: 1.4em;
  }

  .geb2>div {
    align-self: flex-end;
    margin: 0 1em 0 0;
  }

}
