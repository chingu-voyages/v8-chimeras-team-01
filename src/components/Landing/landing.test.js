import React from 'react';
import { shallow } from 'enzyme';

import Landing from './Landing.js';

describe('Landing', () => {

  it('should render correctly', () => {
    const component = shallow(<Landing />);
    expect(component).toMatchSnapshot();
  });

});
