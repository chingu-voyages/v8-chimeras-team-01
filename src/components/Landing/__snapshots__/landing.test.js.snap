// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render correctly 1`] = `
<section
  id="landing pm0"
>
  <i
    className="fas fa-question pointy lrg-scrn-icon"
    onClick={[Function]}
  />
  <section
    className="pm0"
    id="view-button-container"
  >
    <div
      className="button-box fbc sel"
    >
      <i
        className="fas fa-question fa-3x"
        onClick={[Function]}
      />
    </div>
    <div
      className="button-box fbc"
    >
      <i
        className="fas fa-gamepad fa-3x"
        onClick={[Function]}
      />
    </div>
  </section>
  <section
    className="view-container pm0"
  >
    <div
      className="ge-container pm0 shift-up"
    >
      <button
        className="info-close pm0 pointy"
        onClick={[Function]}
      >
         X close 
      </button>
      <div
        className="gameplay-example"
      >
        <div
          className="fbc"
        >
          <h4
            className="ge-header fbc"
          >
            How do I earn Bragging Rights?
          </h4>
        </div>
        <div
          className="ge-box b-s pm0 geb1"
        >
          <div
            className="ge-description b-s fbc"
          >
            Setup a game with a few clicks
          </div>
        </div>
        <div
          className="ge-box b-s pm0 geb2"
        >
          <div
            className="ge-description b-s fbc"
          >
            Answer the questions as they come
          </div>
        </div>
        <div
          className="ge-box b-s pm0 geb3"
        >
          <div
            className="ge-description b-s fbc"
          >
            See who won Bragging Rights
          </div>
        </div>
      </div>
    </div>
    <section
      className="start-game-container pm0 shift-right"
    >
      <div
        className="start-game-section fbc"
      >
        <h4
          className="fbc"
        >
          Quick Launch
        </h4>
        <button
          className="start-game-button fbc pointy"
          onClick={[Function]}
        >
          <p
            className="pm0"
          >
            <strong>
              Are you a GameMaster??
            </strong>
          </p>
          <p
            className="pm0"
          >
            Host A Game
          </p>
        </button>
        <button
          className="start-game-button fbc pointy"
        >
          <Link
            className="link fbc"
            to="/join"
          >
            <p
              className="pm0"
            >
              <strong>
                Have a Code??
              </strong>
            </p>
            <p
              className=" pm0"
            >
              Join A Game
            </p>
          </Link>
        </button>
        <button
          className="browse-button fbc pointy"
          onClick={[Function]}
        >
          <p
            className="pm0"
          >
            Browse Games
          </p>
        </button>
      </div>
      <div
        className="hth-container pm0"
      >
        <div
          className="how-to-host"
        >
          <h4
            className="hth-header pm0 fbc"
          >
            Want to Host a Game?
          </h4>
          <div
            className="how-to-box htb1 b-s"
          >
            <span
              className="htb-num"
            >
              1
            </span>
            <p>
              Choose "GameMaster" Option Above
            </p>
          </div>
          <div
            className="how-to-box htb2 b-s"
          >
            <span
              className="htb-num"
            >
              2
            </span>
            <p>
              Select A Game
            </p>
          </div>
          <div
            className="how-to-box htb3 b-s"
          >
            <span
              className="htb-num"
            >
              3
            </span>
            <p>
              Invite Friends!
            </p>
          </div>
        </div>
      </div>
    </section>
    <Games
      handleToggleGames={[Function]}
      showGames={false}
    />
  </section>
</section>
`;
